### DISCUSSION TITLE
Query parameters with special characters / cannot create variable

### DISCUSSION THREAD

**Discussion by mfrey777** [fastapi/fastapi#8426] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
no code
```

### Description

Hello,

My web front-end generates the following request to the server (JSON:API query)

/comments?filter[post]=1,2&filter[author]=12

I have 2 query parameters "filter[post]" and "filter[author]", but I cannot figure out how to access them, as I cannot create a variable called "filter[post]".

How can I read these query parameters ?

 

### Operating System

Linux

### Operating System Details

Ubuntu 20.04 (WSL)

### FastAPI Version

1.0.0

### Python Version

3.10.2

### Additional Context

no add info

**Comment by hitenvats16** [comment]
You can try to rename filter[post] to filter_post and filter[author] to filter_author. filter[post] superficially means a dictionary with name as filter and post with name as key.

**Comment by NewSouthMjos** [comment, marked as answer]
Here is possible solution:
```
import uvicorn
from fastapi import FastAPI, Request
from fastapi.logger import logger


app = FastAPI()


@app.get('/test')
def router_func2(request: Request):
    # request.query_params is a dict-like
    # You should do this check:
    # if 'filter[post]' in request.query_params.keys():
    #     ...
    # but i'll skip it for simplify
    filter_post = request.query_params['filter[post]']
    filter_author = request.query_params['filter[author]']

    # Now you can access variables filter_post and filter_author
    logger.error(f'{filter_post=} {filter_author=}')
    return f'{filter_post=} {filter_author=}'

if __name__ == "__main__":
    uvicorn.run("main:app", host="0.0.0.0", port=5600, workers=1)
```

logs after access point `/test?filter[post]=1,2&filter[author]=12` :
```
fastapi_dependency  | INFO:     Started server process [1]
fastapi_dependency  | INFO:     Waiting for application startup.
fastapi_dependency  | INFO:     Application startup complete.
fastapi_dependency  | INFO:     Uvicorn running on http://0.0.0.0:5600 (Press CTRL+C to quit)
fastapi_dependency  | filter_post='1,2' filter_author='12'
fastapi_dependency  | INFO:     10.77.78.94:56832 - "GET /test?filter[post]=1,2&filter[author]=12 HTTP/1.1" 200 OK
```
